File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        left=right=0
        n=len(s)
        unique=[0]*26
        maxcount=0
        while right<n:
            unique[ord(s[right])-65]+=1
            while(right-left+1-max(unique) >k):
                unique[ord(s[left])-65]-=1
                left+=1
            maxcount=max(maxcount,right-left+1)
            right+=1
        return maxcount
        

File: Data Structures & Algorithms/max-water-container/submission-0.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int left=0;
        int right=heights.size()-1;
        int capacity=0;
        while(left<right){
            int cap=(right-left)*min(heights[left],heights[right]);
            capacity=max(cap,capacity);
            if(heights[left]>=heights[right]){
                right--;
            }
            else{
                left++;
            }
            
        }
        return capacity;
        
    }
};
